src/App.jsx

```diff
@@ -1,10 +1,15 @@
 import { useState } from 'react'
 import React from 'react'
-import { Table, Tag } from 'antd';
+import { Table, Tag, Input } from 'antd';
 
 
+const { Search } = Input;
+
 function App() {
   const [data, setData] = useState(null);
+  const [filterVal, setFilterVal] = useState("")
+  const [filteredData, setFilteredData] = useState(null);
```

Comment:
I think that empty array is a better solution. You don't have to do checks like `data || []` later in the code every time...

```suggestion
  const [filteredData, setFilteredData] = useState([]);
```

Reply:
makes sense. I didn't realise that before.